lib/palindrome_check.rb

```diff
@@ -1,5 +1,44 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
+# Time: O(n)
+# Space: O(1)
+require 'pry'
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil # must be the first thing
+    return false
+  end
+
+  # recursive way
+
+  # return true if (my_phrase.length == 1 || my_phrase.length == 0)
+  # if my_phrase[0] != my_phrase[my_phrase.length - 1]
+  #   return false
+  # else
+  #   return palindrome_check(my_phrase[1..my_phrase.length - 2])
+  # end
+
+
+  if my_phrase.length == 0 || my_phrase.length == 1
+    return true
+  end
+
+
+  i = 0
+  j = my_phrase.length - 1
+  while i < my_phrase.length
```

Comment:
_i_ does not need to go through all indices from 0 through length-1, It's sufficient to reach mid-way through the string to confirm whether the string is a palindrome or not. (Similar to string_reverse).
So, line 28, could simply be `i < j`.